fix(skills): update Grids w/ Lite and Forms clarification/expand (#141)
IgniteUI/igniteui-react: skills/igniteui-react-components/reference/CHARTS-GRIDS.md

/* master-view.module.css */
.grid-lite {
  min-width: 400px;
  height: 600px;
  flex-grow: 1;
  flex-basis: 0;
}
```

## Grid Lite Example with Column Configurations and Templates

Use `IgrGridLiteColumn` to define columns explicitly with typed `dataType` and optional `cellTemplate` for custom rendering. Set `autoGenerate={false}` (or omit it) when providing explicit columns.

```tsx
import { IgrGridLite, IgrGridLiteColumn, type IgrCellContext } from 'igniteui-react/grid-lite';
import styles from './order-list.module.css';

interface Order {
  id: number;
  customer: string;
  total: number;

>   date: string;

Reviewer: ```suggestion
  date: Date;
```

  status: 'pending' | 'shipped' | 'delivered';
}

const orders: Order[] = [
  { id: 1, customer: 'Alice', total: 149.99, date: '2024-03-01', status: 'delivered' },
  { id: 2, customer: 'Bob', total: 89.50, date: '2024-03-10', status: 'shipped' },

>   { id: 3, customer: 'Carol', total: 220.00, date: '2024-03-15', status: 'pending' },

Reviewer: ```suggestion
  { id: 1, customer: 'Alice', total: 149.99, date: new Date('2024-03-01'), status: 'delivered' },
  { id: 2, customer: 'Bob', total: 89.50, date: new Date('2024-03-10'), status: 'shipped' },
  { id: 3, customer: 'Carol', total: 220.00, date: new Date('2024-03-15'), status: 'pending' },
```

];

> // Simple cell templates — render JSX based on the cell value or row data
> const currencyTemplate = (ctx: IgrCellContext<Order>) => (
>   <span>${(ctx.value as number).toFixed(2)}</span>
> );

Reviewer: ```suggestion
// Simple cell templates — render JSX based on the cell value or row data
const currencyTemplate = (ctx: IgrCellContext<Order>) => (
  <span>${(ctx.value as number).toFixed(2)}</span>
);

const dateTemplate = (ctx: IgrCellContext<Order>) => (
  <span>${(ctx.value as Date).toLocaleDateString()}</span>
);
```

const statusTemplate = (ctx: IgrCellContext<Order>) => {
  const colors: Record<Order['status'], string> = {
    pending: 'orange',
    shipped: 'blue',
    delivered: 'green',
  };
  return <span style={{ color: colors[ctx.value as Order['status']] }}>{ctx.value}</span>;
};

export default function OrderList() {
  return (
    <div className={styles['grid-lite']}>
      <IgrGridLite data={orders}>
        {/* dataType ensures correct sorting and filtering behavior */}
        <IgrGridLiteColumn field="id" dataType="number" />
        <IgrGridLiteColumn field="customer" dataType="string" />
        <IgrGridLiteColumn field="date" dataType="date" />

>         {/* Columns with custom cell templates */}

Reviewer: ```suggestion
        {/* Columns with custom cell templates */}
        <IgrGridLiteColumn field="date" cellTemplate={dateTemplate} />
```

        <IgrGridLiteColumn field="total" dataType="number" cellTemplate={currencyTemplate} />
        <IgrGridLiteColumn field="status" dataType="string" cellTemplate={statusTemplate} />
      </IgrGridLite>
    </div>
  );
}
```

```css
/* order-list.module.css */
.grid-lite {
  min-width: 400px;
  height: 600px;
  flex-grow: 1;
  flex-basis: 0;
}
```

> **Column configuration notes:**

> > - `dataType` accepts `"string"`, `"number"`, `"boolean"`, or `"date"` — set it explicitly so sorting and filtering work correctly for each column type.

Reviewer: ```suggestion
> - `dataType` accepts `"string"` (default), `"number"`, `"boolean"` — set it explicitly so sorting and filtering work correctly for each column type.
```
